### DISCUSSION TITLE
Setup on Unraid Server

### DISCUSSION THREAD

**Discussion by jvanvolken** [inventree/InvenTree#2438] [Q&A] [answered]
Howdy, InvenTree team!

I'm having trouble setting this up on an Unraid server. I was somewhat able to install it from Docker Hub through Unraid's Community Applications with limited customizability.
![image](https://user-images.githubusercontent.com/43481443/145333790-5ef6ee5a-2524-42a0-96b2-320ddaca193f.png)

During the initial setup, I was unable to provide a port or set any typical settings I'm accustomed to seeing. 
![image](https://user-images.githubusercontent.com/43481443/145333517-4dc93ff1-c94e-4d68-9077-c868874dab3f.png)

This gave me an InvenTree application in my list of Dockers without any port mappings.
![image](https://user-images.githubusercontent.com/43481443/145333438-30a2c00c-a3a5-451e-b827-01f590879682.png)

From Unraid, I can access InvenTree's console but I've got no clue how to use it. 
![image](https://user-images.githubusercontent.com/43481443/145333627-487d909e-733e-419b-96a9-d66945ad3312.png)

And of course, to no-one's surprise, going to http://localhost:8000 doesn't work.

Any help is much appreciated! 
Thanks!

**Comment by matmair** [comment, marked as answer]
Hi @jvanvolken !
As far as I understand unraids CA / docker feature this is not going to work this way. You actually need a database(postgres or mysql), a inventree-server, a inventree-worker and a static webserver/proxy instance for a full production setup. You can see the docker-compose config [in the repo](https://github.com/inventree/InvenTree/blob/2b27fab77150fe575aceb159575a23c16c3605b4/docker/docker-compose.yml).
My understanding is that unraid does not work with compose files so you will need to set this structure up yourself. If you come up with something automatical or write a tutorial feel free to send us a link or PR that we can include in the deploy docs.
Regards, Matthias

**Reply by jvanvolken** [reply]
Ah, dang. 

I _do_ already have Mariadb setup, you can see it in one of the images above... so I've got that going for me I guess. As for docker-compose, there was some talk about Unraid supporting it, but man is it confusing for a novice like myself. 

It's a shame that it's not a simple docker install. Not many applications look quite like Inventree.

**Reply by matmair** [reply]
@jvanvolken I mean you can basically just create the instances that are listed in the compose file and set the settings in the env file. That should run without problems.
It is rather the norm to use this kind of deployment method as modern applications need a lot of services. I am relatively new to this stuff myself but I highly recommend learning about it. It is truly powerful and coupled with k8s or similar things can scale your applications extermly well.

**Reply by matmair** [reply]
I am not really sure if we can solve this. We do not support this deployment method and I am not sure if we will in the future. Maybe mark this as an answer for ppl. who search for this.

**Reply by SchrodingersGat** [reply]
As @matmair has outlined, if you can replicate what the `.docker-compose` file does, with environment variables etc, you might be able to get it working. However this is somewhat out of scope of the core InvenTree software :) Everyone's environment is a bit different!

If you can work out how to configure each image, you may be able to get it configured manually.
